src/fusion-order/fusion-extension.ts

```diff
@@ -61,24 +66,444 @@ export class FusionExtension {
+     *
+     * 1 byte - flags:
+     *      01 bit `CUSTOM_RECEIVER_FLAG` - set to 1 if order has custom receiver
+     * 20 bytes — integrator fee recipient
+     * 20 bytes - protocol fee recipient
+     * [20 bytes] — receiver of taking tokens (optional, if not set, maker is used). See `CUSTOM_RECEIVER_FLAG` flag
+     * Same as in `buildAmountGetterData`
+     * @see buildAmountGetterData
+     * @see https://github.com/1inch/limit-order-protocol/blob/22a18f7f20acfec69d4f50ce1880e8e662477710/contracts/extensions/FeeTaker.sol#L114
+     */
+    private buildInteractionData(): string {
+        const flags = new BN(0n).setBit(
+            FusionExtension.CUSTOM_RECEIVER_FLAG_BIT,
+            Boolean(this.extra?.customReceiver)
+        )
+
+        const integratorReceiver =
+            this.extra?.fees?.integrator.integrator || Address.ZERO_ADDRESS
+        const protocolReceiver =
+            this.extra?.fees?.protocol || Address.ZERO_ADDRESS
+
+        const builder = new BytesBuilder()
+            .addUint8(flags)
+            .addAddress(integratorReceiver.toString())
+            .addAddress(protocolReceiver.toString())
+
+        if (this.extra?.customReceiver) {
+            builder.addAddress(this.extra?.customReceiver.toString())
+        }
+
+        builder.addBytes(this.buildAmountGetterData(false))
+
+        return builder.asHex()
+    }
+
+    /**
+     * Build data for getMakingAmount/getTakingAmount
+     *
+     * AuctionDetails
+     * 2 bytes — integrator fee percentage (in 1e5)
+     * 1 byte - integrator share percentage (in 1e2)
+     * 2 bytes — resolver fee percentage (in 1e5)
+     * 1 byte - whitelist discount numerator (in 1e2)
+     * Whitelist
+     *
+     * @see https://github.com/1inch/limit-order-settlement/blob/82f0a25c969170f710825ce6aa6920062adbde88/contracts/SimpleSettlement.sol#L34
+     */
+    private buildAmountGetterData(withAuction: boolean): string {
+        const builder = new BytesBuilder()
+
+        if (withAuction) {
+            this.auctionDetails.encodeInto(builder)
+        }
+
+        const integrator = {
+            fee:
+                this.extra?.fees?.integrator.fee.toFraction(Fees.BASE_1E5) || 0,
+            share:
+                this.extra?.fees?.integrator.share.toFraction(Fees.BASE_1E2) ||
+                0
+        }
+
+        const resolverFee =
+            this.extra?.fees?.resolver.fee.toFraction(Fees.BASE_1E5) || 0
+        const whitelistDiscount =
+            this.extra?.fees?.resolver.whitelistDiscount || Bps.ZERO
+
+        builder
+            .addUint16(BigInt(integrator.fee))
+            .addUint8(BigInt(integrator.share))
+            .addUint16(BigInt(resolverFee))
+            .addUint8(
+                BigInt(
+                    // contract expects discount numerator, but class contain discount
+                    Number(Fees.BASE_1E2) -
+                        whitelistDiscount.toFraction(Fees.BASE_1E2)
+                )
+            )
```

Comment:
I think in case of zero integrator fee and resolver fee we can set always 0 to decrease calldata gas costs